deeplearning4j/deeplearning4j-nlp-parent/deeplearning4j-nlp/src/main/java/org/deeplearning4j/iterator/BertIterator.java

```diff
@@ -299,6 +354,52 @@ public MultiDataSet next(int num) {
         return new Pair<>(outLength, tokenizedSentences);
     }
 
+    private Pair<long[], List<Pair<List<String>, String>>> tokenizePairsMiniBatch(List<Triple<String, String, String>> listPairs) {
+        List<Pair<List<String>, String>> tokenizedSentences = new ArrayList<>(listPairs.size());
+        long[] segIdOnesFrom = new long[listPairs.size()];
+        int idx = 0;
+        for (Triple<String, String, String> t : listPairs) {
+            List<String> tokensL = tokenizeSentence(t.getFirst(), true);
+            List<String> tokensR = tokenizeSentence(t.getSecond(), true);
+            List<String> tokens = new ArrayList<>(maxTokens);
+            int maxLength = maxTokens;
+            if (prependToken != null)
+                maxLength--;
+            if (appendToken != null)
+                maxLength -= 2;
+            if (tokensL.size() + tokensR.size() > maxLength) {
+                boolean shortOnL = tokensL.size() < tokensR.size();
+                int shortSize = Math.min(tokensL.size(), tokensR.size());
+                if (shortSize > maxLength / 2) {
+                    //both lists need to be sliced
+                    tokensL.subList(maxLength / 2, tokensL.size()).clear(); //if maxsize/2 is odd pop extra on L side to match implementation in TF
+                    tokensR.subList(maxLength - maxLength / 2, tokensR.size()).clear();
+                } else {
+                    //slice longer list
+                    if (shortOnL) {
+                        //longer on R - slice R
+                        tokensR.subList(tokensR.size() - (maxLength - tokensL.size()), tokensR.size()).clear();
+                    } else {
+                        //longer on L - slice L
+                        tokensL.subList(tokensL.size() - (maxLength - tokensR.size()), tokensL.size()).clear();
+                    }
```

Comment:
This subList implementation slices on the right - i.e., takes last N tokens. I would have thought we'd want to take the first N instead? That would look like `subList(0, <subset length>)`

Reply:
Yeah, caught this and fixed it.

Reply:
This still look off to me. If we're taking the first N tokens, the first index should be 0
Signature of subList is `subList(int fromIndex, int toIndex)`

Reply:
There is a .clear() at the end of that i.e these are the elements popped.

Reply:
Ah, ok, I missed that. Not the way I would have done it - but it works 👍